src/dmd/parse.d

```diff
@@ -2901,6 +2901,13 @@ final class Parser(AST) : Lexer
                     }
                 case TOK.in_:
                     stc = STC.in_;
+                    if (global.params.vin)
+                    {
+                        deprecation("`in` is defined as `scope const`.  However, `in` has not yet been properly " ~
+                            "implemented, so its current implementation is equivalent to `const`. It is recommended " ~
+                            "to avoid using `in`, and explicitly use `scope const` or `const` instead, until `in` " ~
+                            "is properly implemented.");
```

Comment:
I think this is a bit too long for a deprecation message that might be displayed hundreds of times.

Reply:
How about now?

Reply:
An alternative could be to emit a longer version only once, and a shorter message for other occurrences.

Reply:
That sounds like a good idea.  I think I'll try to implement that tomorrow.

Reply:
Probably simplest with a supplemental deprecation message.

Reply:
Done.  Thank you.